src/Microsoft.DotNet.Cli.Utils/CommandResolver.cs

```diff
@@ -0,0 +1,191 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using NuGet.Frameworks;
+using System.Linq;
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using NuGet.Packaging;
+
+namespace Microsoft.DotNet.Cli.Utils
+{
+    internal static class CommandResolver
+    {
+        public static ProcessStartInfo TryResolvePathAndArgs(string commandName, string args, NuGetFramework framework = null)
+        {
+            return ResolveFromRootedCommand(commandName, args) ??
```

Comment:
- Similar to PATH resolution, resolving from a rooted path is machine specific and might lead to "it works on my machine" issues. I'm wondering whether to add rooted path command resolution as an incremental safety check. If not, I'll need to add it to the input files required for incremental compilation